src/Generator/PropertyGenerator.php

```diff
@@ -21,12 +21,40 @@ class PropertyGenerator extends AbstractMemberGenerator
 
     protected bool $isConst = false;
 
+    protected ?TypeGenerator $type = null;
+
     protected ?PropertyValueGenerator $defaultValue = null;
 
     private bool $omitDefaultValue = false;
 
+    /**
+     * @param  PropertyValueGenerator|string|array|null  $defaultValue
+     * @param  int|int[]                                 $flags
+     */
+    public function __construct(
+        ?string $name = null,
+        $defaultValue = null,
+        $flags = self::FLAG_PUBLIC,
+        ?TypeGenerator $type = null
+    ) {
+        parent::__construct();
+
+        if (null !== $name) {
+            $this->setName($name);
+        }
+        if (null !== $defaultValue) {
+            $this->setDefaultValue($defaultValue);
+        }
+        if ($flags !== self::FLAG_PUBLIC) {
+            $this->setFlags($flags);
+        }
+        if (null !== $type) {
```

Comment:
Conditional can be removed, if `setType()` accepts `null` too

Reply:
Remind me again, `setDefaultValue()` can take `null`, and there's a check for `null` in this constructor.

Why do we do it differently here?

Reply:
49d33430a8d09699aa5aed85bcb0b3bfdf7033c1